circuit-benchmarks/src/mpt_circuit.rs

```diff
@@ -0,0 +1,110 @@
+//! MPT circuit benchmarks
+
+#[cfg(test)]
+mod tests {
+    use ark_std::{end_timer, start_timer};
+    use halo2_proofs::{
+        halo2curves::bn256::{Bn256, Fr, G1Affine},
+        plonk::{create_proof, keygen_pk, keygen_vk, verify_proof},
+        poly::{
+            commitment::ParamsProver,
+            kzg::{
+                commitment::{KZGCommitmentScheme, ParamsKZG, ParamsVerifierKZG},
+                multiopen::{ProverSHPLONK, VerifierSHPLONK},
+                strategy::SingleStrategy,
+            },
+        },
+        transcript::{
+            Blake2bRead, Blake2bWrite, Challenge255, TranscriptReadBuffer, TranscriptWriterBuffer,
+        },
+    };
+    use rand::SeedableRng;
+    use rand_xorshift::XorShiftRng;
+    use std::env::var;
+    use zkevm_circuits::mpt_circuit::{witness_row::Node, MPTCircuit};
+
+    #[cfg_attr(not(feature = "benches"), ignore)]
+    #[test]
+    fn bench_mpt_circuit_prover() {
+        let degree: u32 = var("DEGREE")
+            .expect("No DEGREE env var was provided")
+            .parse()
+            .expect("Cannot parse DEGREE env var as u32");
+
+        let path = "../zkevm-circuits/src/mpt_circuit/tests/UpdateOneLevel.json";
+        let file = std::fs::File::open(path.clone());
+        let reader = std::io::BufReader::new(file.unwrap());
+
+        let randomness: Fr = Fr::one() + Fr::one();
+        let nodes: Vec<Node> = serde_json::from_reader(reader).unwrap();
+
+        let mut keccak_data = vec![];
+        for node in nodes.iter() {
+            for k in node.keccak_data.iter() {
+                keccak_data.push(k.clone());
+            }
+        }
+
+        let circuit = MPTCircuit::<Fr> {
+            nodes,
+            keccak_data,
+            randomness,
+        };
+
+        // Initialize the polynomial commitment parameters
+        let mut rng = XorShiftRng::from_seed([
+            0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06,
+            0xbc, 0xe5,
+        ]);
+
+        // Bench setup generation
+        let setup_message = format!("Setup generation with degree = {}", degree);
```

Comment:
Nitpick: the other circuits use `SETUP_PREFIX` etc... for some of these strings.